django/apps/config.py

```diff
@@ -139,7 +139,11 @@ def create(cls, entry):
                 "'%s' must supply a name attribute." % entry)
 
         # Ensure app_name points to a valid module.
-        app_module = import_module(app_name)
+        try:
+            app_module = import_module(app_name)
+        except ImportError:
+            raise ImproperlyConfigured(
+                "Cannot import '%s', check AppConfig.name is correctly defined.")
```

Comment:
To that end, you should use `assertRaisesMessage` in the test to catch that mistake.